rfcs/quic.md

```diff
@@ -0,0 +1,115 @@
+2.  The server needs to be scriptable, and the custom handlers should live in
+    the wpt repo next to test files, so that they can be updated together.
+3.  An async programming model is preferred.
+
+### Architecture
+
+The QUIC server has a standalone CLI, and is managed by `wpt serve` as a
+subprocess. Configurations are stored in `config.json` (same as `wptserve`), and
+`wpt serve` is responsible for passing the relevant configurations (e.g. port
+numbers) to the QUIC server via command line flags. `wpt serve` handles the
+error gracefully if the QUIC server cannot be started.
+
+The QUIC server is written in Python 3 and makes no attempt to support Python 2.
+Its entrypoint will have a shebang that explicitly requires `python3`. An
+implication is that the QUIC server will be relatively self contained, without
+dependencies on other modules in wpt that are currently Python 2-only.
+
+Code for the QUIC server itself will live in `tools/quic`.
+
+### User interface
+
+For **test writers**: the QUIC server will support Python custom handlers
+similar to `wptserve`. The APIs are not yet finalized and will likely be
+different. To more easily distinguish the two, we add a filename flag to QUIC
+custom handlers (`.quic.py`). These handlers will be ignored by `wptserve`.
+Tests that require the QUIC server need to be marked with a meta tag
+(tentatively `<meta name="quic" content="true">` and `// META: quic=true`). This
+information will be extracted by `wpt manifest` into `MANIFEST.json`, so that
+`wptrunner` can skip the tests that require the QUIC server easily. A lint rule
+will be added to require this meta tag if `.quic.py` appears in the file, even
+though it cannot prevent all accidental dependency of the QUIC server (e.g.
+through a resource script).
+
+For **test runners**: the goal is to not introduce any observable changes to
+users not yet concerned with QUIC or WebTransport. All existing tests will
+continue to run without any infra changes, and all new QUIC-dependent tests will
+be skipped by default. To enable the QUIC server, users need to pass a flag to
+`wpt run` (or `wpt serve`). `wpt` front-end will then attempt to download the
+the dependencies (wheels) from PyPI into `_venv`; maintainers of infrastructures
+where PyPI cannot be used need to manage dependencies themselves (see the next
+section).
+
+### Notable dependencies
+
+* [sys] Python 3 (>=3.6)
+* [py] aioquic
+    * [py] cryptography
+        * [sys] openssl (>=1.0.1, already required by [HTTP/2](http2.md))
+    * [py] pylsqpack
+
+All of the Python dependencies above have native extensions, which requires
+users to use a platform supported by the prebuilt wheels from PyPI or build and
+distribute the wheels themselves (e.g. in Chromium through
+[cipd](https://chromium.googlesource.com/chromium/src/+/master/docs/cipd.md)).
+
+The source code of all dependencies will also be vendored into
+`tools/third_party` to allow building from source if needed, even though we
```

Comment:
Remove everything after the comma (there's a typo, but also I don't think there's any sense in talking about whether there's a plan for that in the wpt infra; this is for vendors).

Reply:
Done.